translator/trace/zipkinv1_thrift_to_protospan.go

```diff
@@ -0,0 +1,211 @@
+			Endpoint:  toTranslatorEndpoint(ztAnnot.Host),
+		}
+		annotations = append(annotations, annot)
+	}
+	return parseZipkinV1Annotations(annotations)
+}
+
+func toTranslatorEndpoint(e *zipkincore.Endpoint) *endpoint {
+	if e == nil {
+		return nil
+	}
+
+	var ipv4, ipv6 string
+	if e.Ipv4 != 0 {
+		ipv4 = net.IPv4(byte(e.Ipv4>>24), byte(e.Ipv4>>16), byte(e.Ipv4>>8), byte(e.Ipv4)).String()
+	}
+	if len(e.Ipv6) != 0 {
+		ipv6 = net.IP(e.Ipv6).String()
+	}
+	return &endpoint{
+		ServiceName: e.ServiceName,
+		IPv4:        ipv4,
+		IPv6:        ipv6,
+		Port:        int32(e.Port),
+	}
+}
+
+var trueByteSlice = []byte{1}
+
+func zipkinV1ThriftBinAnnotationsToOCAttributes(ztBinAnnotations []*zipkincore.BinaryAnnotation) (attributes *tracepb.Span_Attributes, localComponent string) {
+	if len(ztBinAnnotations) == 0 {
+		return nil, ""
+	}
+
+	attributeMap := make(map[string]*tracepb.AttributeValue)
+	for _, binaryAnnotation := range ztBinAnnotations {
+		pbAttrib := &tracepb.AttributeValue{}
+		binAnnotationType := binaryAnnotation.AnnotationType
+		switch binaryAnnotation.AnnotationType {
+		case zipkincore.AnnotationType_BOOL:
+			vBool := bytes.Equal(binaryAnnotation.Value, trueByteSlice)
+			pbAttrib.Value = &tracepb.AttributeValue_BoolValue{BoolValue: vBool}
+		case zipkincore.AnnotationType_BYTES:
+			bytesStr := base64.StdEncoding.EncodeToString(binaryAnnotation.Value)
+			pbAttrib.Value = &tracepb.AttributeValue_StringValue{
+				StringValue: &tracepb.TruncatableString{Value: bytesStr}}
+		case zipkincore.AnnotationType_DOUBLE:
+			var d float64
+			if err := bytesToNumber(binaryAnnotation.Value, &d); err != nil {
+				pbAttrib.Value = strAttributeForError(err)
+			} else {
+				pbAttrib.Value = &tracepb.AttributeValue_DoubleValue{DoubleValue: d}
+			}
+		case zipkincore.AnnotationType_I16:
+			var i int16
+			if err := bytesToNumber(binaryAnnotation.Value, &i); err != nil {
+				pbAttrib.Value = strAttributeForError(err)
+			} else {
+				pbAttrib.Value = &tracepb.AttributeValue_IntValue{IntValue: int64(i)}
+			}
+		case zipkincore.AnnotationType_I32:
+			var i int32
+			if err := bytesToNumber(binaryAnnotation.Value, &i); err != nil {
+				pbAttrib.Value = strAttributeForError(err)
+			} else {
+				pbAttrib.Value = &tracepb.AttributeValue_IntValue{IntValue: int64(i)}
+			}
+		case zipkincore.AnnotationType_I64:
+			var i int64
+			if err := bytesToNumber(binaryAnnotation.Value, &i); err != nil {
+				pbAttrib.Value = strAttributeForError(err)
+			} else {
+				pbAttrib.Value = &tracepb.AttributeValue_IntValue{IntValue: i}
+			}
+		case zipkincore.AnnotationType_STRING:
+			pbAttrib.Value = &tracepb.AttributeValue_StringValue{
+				StringValue: &tracepb.TruncatableString{Value: string(binaryAnnotation.Value)}}
+		default:
+			err := fmt.Errorf("unknown zipkin v1 binary annotation type (%d)", int(binAnnotationType))
+			pbAttrib.Value = strAttributeForError(err)
+		}
+
+		key := binaryAnnotation.Key
+		if key == zipkincore.LOCAL_COMPONENT {
```

Comment:
Opened https://github.com/census-instrumentation/opencensus-service/issues/352 so the same behavior regarding LOCAL_COMPONENT is also added to zipkin v1.